### PR TITLE
Adds a Ranked list input modal

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

Adds a Ranked list input modal. It is similar to a tgui list select modal, but instead of selecting one element, the user orders them by preference. When they click submit the order they are in will be set as the result.

## Why It's Good For The Game

Makes it a lot easier to take in ranked user inputs. For example, first past the post map voting, or choosing their ERT prefs. (Not implemented in this PR.)

## Images of changes

Current version has a little drag handle near each option
image

https://github.com/user-attachments/assets/4da20536-43b4-43ba-9b57-59af67d2c08d

image
image

## Testing

<hr>
See Above

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

NPFC

### CODE REVIEW

File: code/modules/tgui/tgui_input/ranked_list_input.dm [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		var/client/client = user
+		user = client.mob
+
+	if(isnull(user.client))
+		return
+
+	// doesn't support this
+	// if(user.client?.prefs?.toggles2 & PREFTOGGLE_2_DISABLE_TGUI_INPUT)
+	// 	return input(user, message, title, default) as null|anything in items
+
```
Comment (lewcc) [inline code/modules/tgui/tgui_input/ranked_list_input.dm:32]: commented out code here

File: code/__HELPERS/lists.dm [hunk 2]
```diff
@@ -920,3 +920,10 @@
 	while(islist(result))
 		result = pickweight(fill_with_ones(result))
 	return result
+
+/**
+ * Checks to make sure that the lists have the exact same contents, ignores the order of the contents.
+ */
+/proc/lists_equal_unordered(list/list_one, list/list_two)
+	// This ensures that both lists contain the same elements by checking if the difference between them is empty in both directions.
+	return !length(list_one - list_two) && !length(list_two - list_one)
```
Comment (lewcc) [inline code/__HELPERS/lists.dm:929]: Do we have the xor operator? This works fine too, just curious

**Review by lewcc — APPROVED** [review]
LGTM, don't have an amazing grip on TGUI but things look fine

**Comment by S34NW** [comment]
Deconflict for merge

**Pull request merged**